pkg/service/store/store.go

```diff
@@ -165,6 +171,135 @@ func NewClusterService(name, namespace string) ClusterService {
 	}
 }
 
+// MCSAPIServiceSpec is similar as ServiceImportSpec except for the fields: IPs,
+// ExportCreationTimestamp and ServiceName.
+//
+// +k8s:deepcopy-gen=true
+type MCSAPIServiceSpec struct {
+	// ServiceName is the name of the ServiceImport or ServiceExport parent resources.
+	ServiceName string `json:"serviceName"`
+	// ExportCreationTimestamp is a timestamp representing the server time when this object was
+	// created. It is not guaranteed to be set in happens-before order across separate operations.
+	// Clients may not set this value. It is represented in RFC3339 form and is in UTC.
+	//
+	// Populated by the system.
+	// Read-only.
+	// Null for lists.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	ExportCreationTimestamp metav1.Time `json:"exportCreationTimestamp"`
+	// +listType=atomic
+	Ports []mcsapiv1alpha1.ServicePort `json:"ports"`
+	// Type defines the type of this service.
+	// Must be ClusterSetIP or Headless.
+	// +kubebuilder:validation:Enum=ClusterSetIP;Headless
+	Type mcsapiv1alpha1.ServiceImportType `json:"type"`
+	// Supports "ClientIP" and "None". Used to maintain session affinity.
+	// Enable client IP based session affinity.
+	// Must be ClientIP or None.
+	// Defaults to None.
+	// Ignored when type is Headless
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+	SessionAffinity v1.ServiceAffinity `json:"sessionAffinity,omitempty"`
+	// SessionAffinityConfig contains session affinity configuration.
+	SessionAffinityConfig *v1.SessionAffinityConfig `json:"sessionAffinityConfig,omitempty"`
+}
+
+// Marshal returns the MCS-API Service Spec object as JSON byte slice
+func (s *MCSAPIServiceSpec) Marshal() ([]byte, error) {
+	return json.Marshal(s)
+}
+
+// Unmarshal parses the JSON byte slice and updates the MCS-API Service Spec receiver
+func (s *MCSAPIServiceSpec) Unmarshal(_ string, data []byte) error {
+	newMCSAPIServiceSpec := MCSAPIServiceSpec{}
+
+	if err := json.Unmarshal(data, &newMCSAPIServiceSpec); err != nil {
+		return err
+	}
+
+	*s = newMCSAPIServiceSpec
+
+	return nil
+}
+
+// DeepEqual is a deepequal function, deeply comparing the receiver with other.
+// in must be non-nil. This is not auto-generated because the auto-generated
+// function doesn't compile because of some fields that we don't control.
+// Although the auto-generated version was taken as a base.
+func (in *MCSAPIServiceSpec) DeepEqual(other *MCSAPIServiceSpec) bool {
```

Comment:
I guess we want to remove the `deepcopy-gen=true` above?
Related to my comment about embedding outside k8s objects. Wouldn't it be better if `MCSAPIServiceSpec` would be standalone and we'd be able to generate the `DeepEqual` for it?

Reply:
> I guess we want to remove the deepcopy-gen=true above?

The deepcopy generation is necessary for the deepcopy of the `ClusterService` struct to work. And for the DeepEqual there is 3 issues: with the generated code:
- It tries to call `DeepEqual` on the `ExportCreationTimestamp` field which doesn't exist (which for reference is a `metav1.Time` object)
- It tries to call `DeepEqual` on the `mcsapiv1alpha1.ServicePort` which doesn't exists
- It tries to call `DeepEqual` on the `SessionAffinityConfig` (which for reference is a `corev1.SessionAffinityConfig` object)

I am not sure how to resolve those without having to implement the function without the code generation unfortunately :/